### ISSUE TITLE
Using bang operator with a property named "type" causes errors

### ISSUE THREAD

**Issue by johnfn** [Microsoft/TypeScript#20604]
**TypeScript Version:** 2.7.0-dev.20171209

**Code**

```ts
class Foo {
    type!: number;
}
```

**Expected behavior:**

No errors.

**Actual behavior:**

I get a number of errors, including "Cannot find name 'type'", an unexpected token error on the `!`, etc. 

This does not happen if I change `type` to any other word.

P.S. I've found I can work around this like so

```ts
class Foo {
    "type": number;
}
```

**Comment by andy-ms** [comment]
Looks like a parser bug.
